hooks/useFetchProjects.ts

```diff
@@ -94,7 +126,52 @@ const useFetchProjects = () => {
     fetchProjects();
   }, [projects, setProjects]);
 
-  return { projects, loading };
+  // handle tags extraction
+  useEffect(() => {
+    const newTags = extractTagsFromProjects(projects?.items?.items);
+    setTags(newTags);
+  }, [projects?.items?.items]);
+
+  // handle projects filtering by tags
+  useEffect(() => {
+    const projectsToDisplay = filterProjectsByTags(
+      projects?.items?.items,
+      tagToFilterBy
+    );
+    setFilteredProjects(projectsToDisplay);
+  }, [tags, tagToFilterBy, projects?.items?.items]);
+
+  // handle projects filtering by tags
+  useEffect(() => {
+    if (tags?.length) {
+      const newFiteredTags = structuredClone(tagToFilterBy);
+      tags.forEach(tag => {
+        const existingTag = newFiteredTags.find(t => t.name === tag);
+        if (!existingTag) {
+          newFiteredTags.push({ name: tag, isActive: true });
+        } else {
+          existingTag.isActive = !existingTag.isActive;
+        }
+      });
+      setTagToFilterBy(n => newFiteredTags);
+    }
+  }, [tags]);
+
+  const toggleTagIsActive = (tag: ProjectFilter) => {
+    const newTags = structuredClone(tagToFilterBy);
+    const existingTag = newTags.find(t => t.name === tag.name);
+    if (existingTag) {
+      existingTag.isActive = !existingTag.isActive;
+      setTagToFilterBy(n => newTags);
+    }
```

Comment:
Same question as before, will this work as `setTagToFilterBy(newTags);` ?

Reply:
this fn takes in a tag and updates the filteredtags state.
`setTagToFilterBy` sets the new state.

Reply:
Ok, but in the `setTagToFilterBy(n => newTags);` part. 
I checked locally and replaced it with `setTagToFilterBy(newTags);` will work the same way. 

So any reason to keep the non used variable inside the arrow function?

Am I missing something here?

Reply:
Got it